extension/datadogextension/extension.go

```diff
@@ -313,6 +319,18 @@ func newExtension(
 	logComponent := agentcomponents.NewLogComponent(set.TelemetrySettings)
 	serializer := agentcomponents.NewSerializerComponent(configComponent, logComponent, host.Identifier)
 
+	// Collect resource attributes from TelemetrySettings.Resource
+	// Format: ["key:value", ...]
+	resourceList := []string{}
+	if attrs := set.Resource.Attributes(); attrs.Len() > 0 {
+		attrs.Range(func(k string, v pcommon.Value) bool {
+			resourceList = append(resourceList, k+":"+v.Str())
+			return true
+		})
```

Comment:
We can use more modern syntax:

```suggestion
		for k, v := range attrs.Range {
			resourceList = append(resourceList, k+":"+v.Str())
		}
```